libevmasm/Assembly.cpp

```diff
@@ -436,7 +436,17 @@ map<u256, u256> Assembly::optimiseInternal(
 			BlockDeduplicator dedup{m_items};
 			if (dedup.deduplicate())
 			{
-				tagReplacements.insert(dedup.replacedTags().begin(), dedup.replacedTags().end());
+				for (auto const& replacement: dedup.replacedTags())
+				{
+					tagReplacements[replacement.first] = replacement.second;
```

Comment:
Should this assert that `replacement.first` doesn't exist yet?

Reply:
We can try and see what happens...